dipy/io/peaks.py

```diff
@@ -0,0 +1,237 @@
+from __future__ import division, print_function, absolute_import
+
+import os
+import numpy as np
+
+from dipy.core.sphere import Sphere
+from dipy.direction.peaks import PeaksAndMetrics
+from distutils.version import LooseVersion
+
+# Conditional import machinery for pytables
+from dipy.utils.optpkg import optional_package
+
+# Allow import, but disable doctests, if we don't have pytables
+tables, have_tables, _ = optional_package('tables')
+
+# Useful variable for backward compatibility.
+if have_tables:
+    TABLES_LESS_3_0 = LooseVersion(tables.__version__) < "3.0"
+
+from dipy.data import get_sphere
+from dipy.core.sphere import Sphere
+
+
+def _safe_save(f, group, array, name):
+    """ Safe saving of arrays with specific names
+
+    Parameters
+    ----------
+    f : HDF5 file handle
+    group : HDF5 group
+    array : array
+    name : string
+    """
+
+    if TABLES_LESS_3_0:
+        func_create_carray = f.createCArray
+    else:
+        func_create_carray = f.create_carray
+
+    if array is not None:
+        atom = tables.Atom.from_dtype(array.dtype)
+        ds = func_create_carray(group, name, atom, array.shape)
+        ds[:] = array
+
+
+def load_peaks(fname, verbose=False):
+    """ Load a PeaksAndMetrics HDF5 file (PAM5)
+
+    Parameters
+    ----------
+    fname : string
+        Filename of PAM5 file.
+    verbose : bool
+        Print summary information about the loaded file.
+
+    Returns
+    -------
+    pam : PeaksAndMetrics object
+    """
+
+    if os.path.splitext(fname)[1] != '.pam5':
+        raise IOError('This function supports only PAM5 (HDF5) files')
+
+    if TABLES_LESS_3_0:
+        func_open_file = tables.openFile
+    else:
+        func_open_file = tables.open_file
+
+    f = func_open_file(fname, 'r')
+
+    pam = PeaksAndMetrics()
+
+    pamh = f.root.pam
+
+    version = f.root.version[0].decode()
+
+    try:
+        affine = pamh.affine[:]
+    except tables.NoSuchNodeError:
+        affine = None
+
+    peak_dirs = pamh.peak_dirs[:]
+    peak_values = pamh.peak_values[:]
+    peak_indices = pamh.peak_indices[:]
+
+    try:
+        shm_coeff = pamh.shm_coeff[:]
+    except tables.NoSuchNodeError:
+        shm_coeff = None
+
+    sphere_vertices = pamh.sphere_vertices[:]
+
+    try:
+        odf = pamh.odf[:]
+    except tables.NoSuchNodeError:
+        odf = None
+
+    pam.affine = affine
+    pam.peak_dirs = peak_dirs
+    pam.peak_values = peak_values
+    pam.peak_indices = peak_indices
+    pam.shm_coeff = shm_coeff
+    pam.sphere = Sphere(xyz=sphere_vertices)
+    pam.B = pamh.B[:]
+    pam.total_weight = pamh.total_weight[:][0]
+    pam.ang_thr = pamh.ang_thr[:][0]
+    pam.gfa = pamh.gfa[:]
+    pam.qa = pamh.qa[:]
+    pam.odf = odf
+
+    f.close()
+
+    if verbose:
+        print('PAM5 version')
+        print(version)
```

Comment:
I think that we should test up-front for the version, and raise an error if a file with the wrong version(s) is provided.